src/main/java/org/codehaus/mojo/license/api/DefaultThirdPartyTool.java

```diff
@@ -837,4 +854,10 @@ private File resolveArtifact( String groupId, String artifactId, String version,
         }
         return migrateKeys;
     }
+
+    @Override
+    public void setAetherRepoSession( RepositorySystemSession aetherRepoSession )
+    {
+        this.aetherRepoSession = aetherRepoSession;
```

Comment:
Making aetherRepoSession mutable in this way does not feel good. If there is no way to persuade plexus to inject it, I vote for removing the aetherRepoSession field and pass it via method parameter.

Reply:
I agree, it is not nice. I couldn't find a way to inject it. I'll move it to method parameters.

Reply:
Replaced with MavenSession injection